### DISCUSSION TITLE
quickFilters : Profile name on Pasted/Cloned filters

### DISCUSSION THREAD

**Discussion by TonyGravagno** [RealRaven2000/FiltaQuilla#282] [Ideas]
When we cut/copy+paste, or clone a filter in quickFilters, an exact copy is saved. What do you think about the following Optional enhancement?

Prefix or Suffix the destination account/profile name to the new filter, separated by a specified delimiter:  
`Site1 Admin - My filter1`  
`Site2 Admin - My filter1`
`My filter1 : Site3 Admin`  
`My filter1 : Site4 Admin`

If the current account/profile name is already on the name, or perhaps if the delimiter already exists, replace the existing name with the new one. Example:

`Site1 Admin - My filter1` : Delimiter is "-", and the operation is prefix.  
There is a dash, and there is text before the dash, so remove that existing text to get `My filter1`.  
Now perform the standard name append operation to get:  
`Site2 Admin - My filter1`

Of course in that scenario you wouldn't want to use a delimiter that already clashes with others, like "my - filter - 1", so you'd use a unique delimiter like colon ":" as in the first examples.

**Comment by RealRaven2000** [comment]
it looks like a feature request for quickFilters. should we open a discussion section in quickFilters too? Btw there are already (some naming rules for new filters, so it would make sense to extend there for clones)

**Reply by TonyGravagno** [reply]
>  naming rules for new filters

Oh? I created a bunch of filters yesterday and was wondering about this.

**Reply by RealRaven2000** [reply]
> > naming rules for new filters
> 
> Oh? I created a bunch of filters yesterday and was wondering about this.

![image](https://github.com/user-attachments/assets/eb8cf933-468a-468d-a128-246ff3705f0b)

hope you using the assistant!

**Reply by TonyGravagno** [reply]
Ahh, yes, appreciate the suffix on cloning.

What I'm proposing here is a dynamic naming rule, where, rather than appending a string like "(copy)", the entire text is a substitution string which the user can order as desired. Examples:

- $profile : The account "prettyname"
- $filter : The filter name

This allows for preferences like "$profile : $filter", "$filter > $profile", etc.

This is similar to the common feature for filenames, like $YY-$MM-$DD ...

The scenario I'm trying to address is the desire/need to manually change the name of every filter as it's cloned and from one account to another. One reason for this is to avoid name conflict issues with filters between accounts. That specific reason might not be forever valid - as soon as that bug is fixed then unique naming between accounts becomes more of a choice than a temporary pain. But given that this is Tb, that situation might never change.

Another reson for per-account filter names is that when filters are copied between accounts, it's impossible to know if they have been tuned for the account (with the account's domain or whatever) without opening it. Having the profile name in the filter name confirms the target account for a given filter (of course once the terms and actions have been changed).

Here is another way to imagine this naming feature: Assume the pretty name on the account is "FooCorp Support". A filter is created as "FooCorp Support : Existing Clients". The filter is cloned to the profile with prettyname "FooCorp Accounting". We know that the text of the source profile name is "FooCorp Support" and that this text is found in the filter name. So when it's cloned that text can be logically substituted with the prettyname of the target profile.

Thoughts?

**Comment by RealRaven2000** [comment]
That may be worth creating a feature request (don't make it too complicated initially) but I do like the thought of placeholder strings in naming rules / why not? We can always add more keywords later, just add the most useful ones. (I will have a tought about what the exact syntax should be - I am already using the `%variable%` `%variable(parameter)%` syntax in custom filter templates (and in my other Add-on SmartTemplates, where this is originally from), so I might simply go with that one.
